### PR TITLE
DOC user guide on the hub

### PR DESCRIPTION
**Pull request by adrinjalali**
A little bit of user guide and improvements to docs.

### CODE REVIEW

**Comment by adrinjalali** [comment]
Ready for review @skops-dev/maintainers

File: docs/index.rst [hunk 1]
```diff
@@ -6,17 +6,31 @@
 Welcome to skops's documentation!
 =================================
 
+``skops`` is a Python library helping you share you `scikit-learn
```
Comment (BenjaminBossan) [inline docs/index.rst:9]: How about linking `skops` to github here? AFAICT, there is no link to the repo anywhere in the documentation at all.

**Review by BenjaminBossan — APPROVED** [review]
Great addition, thanks. I proposed a change, but it's not critical at all.

File: docs/index.rst (hunk 1 above)
Comment (adrinjalali) [reply docs/index.rst:9]: Done!

**Comment by adrinjalali** [comment]
Also wondering what @osanseviero thinks about this.

**Comment by BenjaminBossan** [comment]
> Also wondering what osanseviero thinks about this.

Okay, I'll wait before merging then.

File: docs/hf_hub.rst [hunk 2]
```diff
@@ -0,0 +1,72 @@
+.. _hf_hub:
+
+scikit-learn Models on HuggingFace Hub
+======================================
+
+This library allows you to initialize and create a model repository compatible
+with HuggingFace Hub, which you can push to the hub and call an API to get the
```
Comment (osanseviero) [inline docs/hf_hub.rst:7]: I think you could also mention other benefits apart from the API
* A widget to try the model directly in the browser
* Metadata tags for discoverability
* Collaborating with others in a model
* Downloading/persisting in a way easy to share

File: docs/index.rst [hunk 3]
```diff
@@ -6,17 +6,33 @@
...
```
Comment (osanseviero) [inline docs/index.rst:9]:
```suggestion
``skops`` is a Python library helping you share your `scikit-learn
```

**Review by osanseviero — APPROVED** [review]
Looks good! I just left minor suggestions. It would be great if @merveenoyan could take a quick look on Monday as well unless we're in a hurry to merge this

**Comment by osanseviero** [comment]
Very friendly ping @merveenoyan as I realized last ping was during holidays+before weekend, so I wanted to resurface this.

File: docs/hf_hub.rst [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+  Hub backend to show them in the widget to test the model when visiting the
+  model's page on the Hub.
+- ``environment``: A list of dependencies that the model requires. These
+  packages must be available on conda-forge and are installed before loading
+  the model.
+- ``model.file``: The file name of the persisted model.
+- ``task``: The task of the model.
+
+You almost never need to create or touch this file manually, and it's created
+when you call :func:`skops.hub_utils.init`.
```
Comment (merveenoyan) [inline docs/hf_hub.rst:79]: Maybe an emphasis on this with bold font would be noice but LGTM 😅 it's one of the main value propositions of skops that's why
image

**Review by merveenoyan — APPROVED** [review]
Left minor comments, feel free to ignore if you don't agree but LGTM

File: docs/hf_hub.rst [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+=======================================
+
+This library allows you to initialize and create a model repository compatible
+with `Hugging Face Hub <https://huggingface.co/models>`__, which among other
+things, gives you the following benefits:
+
+- Inference API to get model output through REST calls
+- A widget to try the model directly in the browser
+- Metadata tags for better discoverability of the model
+- Collaborating with others on a model
```
[the remark this answers is not in the document]
Comment (adrinjalali) [reply docs/hf_hub.rst:13]: through what? I'm not sure how to finish the sentence

[the remark this answers is not in the document]
Comment (merveenoyan) [reply docs/hf_hub.rst:13]: Having repositories on Hub in general and pull requests + discussions.

**Pull request merged**